pkg/build/config.go

```diff
@@ -96,3 +96,9 @@ type Config struct {
 	// ModTimestamp string      `yaml:"mod_timestamp,omitempty"`
 	// GoBinary     string      `yaml:",omitempty"`
 }
+
+type ImageConfig struct {
+	ID          string            `yaml:",omitempty"`
+	Annotations map[string]string `yaml:",omitempty"`
+	Labels      map[string]string `yaml:",omitempty"`
```

Comment:
Should these just go in `Config` alongside `Env` etc.? It seems like a meaningless separation.

The only problem with that is if goreleaser adds a `Labels` or `Annotations` field to its config, we'd have a different meaning. I'm starting to really regret copying goreleaser's config so closely. 😢

Reply:
IMHO, we should separate them from a build because these are related to the image we provide at the end of that build. So, it might cause a bit more confusion if we put them into the Config section. In the current design, users should match the id fields of the builds and images sections to be able to define labels and annotations per image related to that build, WDYT?

Reply:
Oh I see, I was confusing `build.env` (what envs to set when we `go build`) vs `image.env` (what envs to set in the resulting image).

In that case, yeah, I agree we should have a separate image config struct with labels/annotations.

We should also use this to fix #633, with a top-level `defaultImageConfig: { }` option, that gets overridden by more specific `images:` and both get overridden if there's a conflicting `--image-label` specified.

WDYT?

Reply:
Oh yeah, we should apply some kind of prioritization between them, that makes a lot of sense 🙋🏻‍♂️

Here are the prioritization rules:

1. The most important one is --image-label, if there is a conflict between others, we should use the defined one by the --image-label flag.
2. The second important one is the images section within the .ko.yaml, so, if we specify the more specific one inside of that section, we should override the defaultImageConfig.
3. The defaultImageConfig has the lowest priority among others. We should use the values within the defaultImageConfig unless others (images:, --image-label) do not specify any values at all.

Reply:
it should be okay @imjasonh 🙋🏻‍♂️